### ISSUE TITLE
Changa+BH tutorial

### ISSUE THREAD

**Issue by Martin-Rey** [pynbody/tangos#24]
A few issues arose when going through the changa BH tutorial:

- Missing --for in the command "tangos_add_bh tutorial_changa_blackholes". Otherwise, one gets an Unrecognised argument error. This is purely a documentation problem.

- Failure of "tangos_add_bh --for tutorial_changa_blackholes" when run with mpi producing:
```
Traceback (most recent call last):
  File "/Users/mrey/anaconda3/bin/tangos_add_bh", line 11, in <module>
    load_entry_point('tangos', 'console_scripts', 'tangos_add_bh')()
  File "/Users/mrey/Documents/tangos/tangos/scripts/add_bh.py", line 335, in main
    parallel_tasks.launch(run)
  File "/Users/mrey/Documents/tangos/tangos/parallel_tasks/__init__.py", line 54, in launch
    backend.launch(_exec_function_or_server, num_procs, [function, args])
  File "/Users/mrey/Documents/tangos/tangos/parallel_tasks/backends/mpi4py.py", line 55, in launch
    function(*args)
  File "/Users/mrey/Documents/tangos/tangos/parallel_tasks/__init__.py", line 86, in _exec_function_or_server
    function(*args)
  File "/Users/mrey/Documents/tangos/tangos/scripts/add_bh.py", line 331, in run
    timelink_bh(args.sims, session)
  File "/Users/mrey/Documents/tangos/tangos/scripts/add_bh.py", line 308, in timelink_bh
    generate_halolinks(session, fname, pairs)
  File "/Users/mrey/Documents/tangos/tangos/scripts/add_bh.py", line 216, in generate_halolinks
    for ts1, ts2 in parallel_tasks.distributed(pairs):
  File "/Users/mrey/Documents/tangos/tangos/parallel_tasks/jobs.py", line 48, in parallel_iterate
    MessageStartIteration(len(task_list)).send(0)
TypeError: object of type 'generator' has no len()
```

**Comment by apontzen** [comment]
Interesting, did you definitely use `--backend=mpi4py`?

**Comment by Martin-Rey** [comment]
Yes, both through command line argument or a config file. I do not get an error when run without mpi.

The script executes (I assume) normally. This error occurs after the last job form the queue is finished. It looks as if the script tries to run with an empty job queue, hence producing the length error.

I obtained the right BH properties nonetheless, crosschecked against the tutorial video.

**Comment by Martin-Rey** [comment]
Fixed by PR #29
